daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -886,6 +886,1034 @@ def make_compute(sdfg, state, vec_width=1):
         return new_sdfg
 
 
+@op_implementation(op="Conv", name="fpga_tiled")
+class FPGAIm2ColConv_tiled(ONNXForward):
+    # ==================================
+    # Im2Col Convolution based on DaCe Master GEMM
+    # ==================================
+    '''
+    Use tiled 1D-Systolic GEMM implementation Y = W @ X + B (A @ B + C)
+    A: the weights W streamed for Im2Col processing
+    B: the Im2Col converted input X
+    C: the Bias B (if applicable)
+    '''
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        X = in_desc_with_name(node, state, sdfg, "X")
+        W = in_desc_with_name(node, state, sdfg, "W")
+        Y = out_desc_with_name(node, state, sdfg, "Y")
+
+        try:
+            B = in_desc_with_name(node, state, sdfg, "B")
+        except Exception as e:
+            B = None
+
+        image_dims = len(X.shape) - 2
+        num_filters = W.shape[0]
+        num_channels = X.shape[1]
+
+        # only do 2D for now
+        if len(X.shape) != 4 or len(W.shape) != 4:
+            return False
+
+        if node.group != 1:
+            return False
+
+        if num_channels != W.shape[1]:
+            return False
+
+        if node.dilations is not None and (not all(d == 1
+                                                   for d in node.dilations) or
+                                           len(node.dilations) != image_dims):
+            return False
+
+        # Support all same padding
+        if node.pads is not None and (not all(p == node.pads[0]
+                                              for p in node.pads)
+                                      or len(node.pads) != image_dims * 2):
+            return False
+
+        if Y.dtype.veclen != X.dtype.veclen:
+            return False
+
+        if node.pads[0] > W.shape[2] // 2:
+            return False
+
+        # assume square kernel
+        if W.shape[2] != W.shape[3]:
```

Comment:
Why? 
Could you please add here, or in the `forward` method some small comments about that?

Reply:
addressed